src/Renderer/ThreeExtented/GlobeControls.js

```diff
@@ -471,41 +469,39 @@ function GlobeControls(camera, domElement, engine) {
     var update = function() {
 
         if (state === CONTROL_STATE.MOVE_GLOBE) {
-            offset.copy(snapShotCamera.position);
-        } else {
-            // get camera position
-            offset.copy(this.camera.position);
-            // transform to local globeTarget
-            offset.applyMatrix4( globeTarget.matrixWorldInverse );
-        }
-
-        offGT.copy(globeTarget.position);
 
-        if (state === CONTROL_STATE.MOVE_GLOBE) {
+            movingGlobeTarget.copy(this.getCenter()).applyQuaternion(quatGlobe);
+            this.camera.position.copy(snapShotCamera.position).applyQuaternion(quatGlobe);
+            this.camera.up.copy(movingGlobeTarget.clone().normalize());
```

Comment:
The controls are still buggy (even after this PR: try clicking outside the globe), so adding comments explaining what the code is supposed to do will help us fix the problems.